bool hook_open
bool core_open